### PR TITLE
fix _estimator_type for cross_valmultiscore

### LINKED ISSUE #4834: cross_val_multiscore uses wrong cross-validation methods during running SlidingEstimator or GeneralizingEstimator in classification tasks
Based on the source code, **cross_val_multiscore** uses [`is_classifier(estimator)`](https://github.com/mne-tools/mne-python/blob/876192598da7c557dbf46a6dda2ced4b7088d4bb/mne/fixes.py#L1067) to assess the type of the input estimator and then create a cv object. 

[By default](https://github.com/mne-tools/mne-python/blob/master/mne/decoding/base.py#L436), if the input cv value is int (e.g. int(10)), and input estimator is a "classifier", it should create a **StratifiedKFold** cv object. 

### **However,**
when using `SlidingEstimator` or  `GeneralizingEstimator` this kind of "second order" estimators as the input estimator [(e.g. the MNE tutorial here)](http://www.nmr.mgh.harvard.edu/mne/dev/auto_tutorials/plot_sensors_decoding.html#temporal-decoding
), **cross_val_multiscore** can not correctly retrieve the estimator types.

This is because this kind of "second order" estimators do not have the `_estimator_type` attribute. 
This causes **cross_val_multiscore** misinterprets classifiers as regressors and creates an inappropriate cv object. 

By default, it creates a **KFold** (and without shuffling!!) object instead of **StratifiedKFold**.
This inappropriate cv may cause some serious problems in some situations (small sample size, trial has been sorted, ect.).

### PR DESCRIPTION
**Pull request by kingjr**
closes #4834

### CODE REVIEW

File: mne/decoding/tests/test_search_light.py [hunk 1]
```diff
@@ -29,13 +29,16 @@ def test_search_light():
...
+    from sklearn.base import is_classifier
 
     X, y = make_data()
     n_epochs, _, n_time = X.shape
     # init
     assert_raises(ValueError, SlidingEstimator, 'foo')
     sl = SlidingEstimator(Ridge())
+    assert_true(not is_classifier(sl))
     sl = SlidingEstimator(LogisticRegression())
+    assert_true(is_classifier(sl))
```
Comment (larsoner) [inline mne/decoding/tests/test_search_light.py]: Should you also check that the resulting `cv` defaults to `StratifiedKFold` and `KFold` in the two cases, respectively?

Comment (kingjr) [reply mne/decoding/tests/test_search_light.py]: That's a bit hard, cross_val_multiscore is a function not an object, so we can't investigate the way it builds the cv.

**Comment by larsoner** [comment]
Can you make a case that clearly fails if it uses KFold but works with the
stratified version? It would be an indirect test of which it uses but it
should suffice (and it would also test the original failure mode actually).

**Comment by kingjr** [comment]
Here you go, but it's a bit of a complicated test...

File: mne/decoding/tests/test_base.py [hunk 2]
```diff
@@ -231,3 +231,20 @@ def test_cross_val_multiscore():
...
+    # otherwise
+    X = np.random.randn(1000, 3)
+    y = np.r_[np.zeros(500), np.ones(500)]
+    clf = LogisticRegression(random_state=0)
+    reg = LinearRegression()
+    for cross_val in (cross_val_score, cross_val_multiscore):
+        manual = cross_val(clf, X, y, cv=StratifiedKFold(2))
+        auto = cross_val(clf, X, y, cv=2)
+        assert_array_equal(manual, auto)
+        assert_raises(ValueError, cross_val, clf, X, y, cv=KFold(2))
```
Comment (agramfort) [inline mne/decoding/tests/test_base.py]: are you preventing the use of KFold for classifiers? if so this sounds too much.

**Comment by kingjr** [comment]
No, this is just a corner case for kfold=2 with this specific y

**Comment by larsoner** [comment]
Thanks @kingjr
